### ISSUE TITLE
Dependency for React 17

### ISSUE THREAD

**Issue by devlom113** [dowjones/react-dropdown-tree-select#521]
**Describe the bug**
There is a dependency blocking apps built with React 17 from adding the package.

**To Reproduce**
`Could not resolve dependency:`
`peer react@"^16.0.0" from react-dropdown-tree-select@2.5.1`

**Expected behavior**
Just a change in the dependencies should fix it.

**Comment by mrchief** [comment]
You're right but I believe we need to address a few lifecycle warnings before we bump it.

**Comment by ldeveber** [comment]
@devlom113 we were able to get around it by purposefully resolving react to 17. It spews LOTS of warnings in the console, but it will "function" since the old lifecycle has not been completely removed yet. I believe it renders more often than it should but it at least works-ish

```

  "resolutions": {
    "react": "^17.0.1"
  },
```

**Comment by mrchief** [comment]
> I believe it renders more often than it should 

Are you sure it's a react thing and not because you're inadvertently re-rendering due to state changes?

**Comment by analuznoro** [comment]
I have this issue when use npm install but if i use yarn this dependency work fine. 
Is there an estimated date of the update to react 17?

**Comment by lcanavesio** [comment]
Hi @mrchief  ! Any update for the estimated date of the new Release ?

**Comment by fre-ben** [comment]
Wondering as well :)

**Comment by HenryYan2012** [comment]
Looking forward the new Release :)

**Comment by lcanavesio** [comment]
Waiting for the new Release

**Comment by mrchief** [comment]
Is anyone able to send a PR on this?

**Comment by lcanavesio** [comment]
@mrchief Hi. Go ahead
I try it and do a PR if it works for me ok.

**Comment by jdlee022** [comment]
Any news on this?

**Comment by adybka-inovex** [comment]
Would love an update

**Comment by adybka-inovex** [comment]
I came across a solution for NPM without having to migrate to Yarn. I ran the command 
`npm i react-dropdown-tree-select --legacy-peer-deps`

Which allowed the package to be installed. From what I read the end flag was made for react 17 to help solve dependency issues. This however removed a package needed for AG-Grid with gave an error when `npm run build` was ran.  To fix this, I reinstalled ag grid with `npm i ag-grid` and everything worked. Other packages may cause issues and conflicts depending on your project but worth a try.
